spec/helpers/blacklight_helper_spec.rb

```diff
@@ -128,4 +132,32 @@
       expect(helper).to have_received(:head).with(:bad_request)
     end
   end
+
+  describe "#link_back_to_catalog_safe" do
+    let(:blacklight_config) { CatalogController.new.blacklight_config }
+    let(:search_state) { Blacklight::SearchState.new({}, blacklight_config) }
+    let(:search_session) { {} }
+
+    context "with valid parameters" do
+      let(:valid_session) { instance_double(Search) }
+      it "produces a link" do
+        allow(valid_session).to receive(:query_params).and_return(
+          action: "show", controller: "catalog", id: "123"
+        )
+        allow(self).to receive(:current_search_session).and_return(valid_session)
+        expect(link_back_to_catalog_safe).to include("/catalog/123")
```

Comment:
Per discussion, we want to try `expect(helper.link_back_to_catalog_safe)` and setting the allows on the `helper` instead of on `self`.

Note also that this helper initialization may obviate a little of the test setup: https://github.com/pulibrary/orangelight/blob/ace3e1ba6188b85a091eac310adb82b18ac1cfe2/spec/support/controller_level_helpers.rb#L16